compiler/rustc_resolve/src/imports.rs

```diff
@@ -554,6 +555,40 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         }
     }
 
+    pub(crate) fn check_local_binding_shadows_glob_reexport(&mut self) {
+        for module in self.arenas.local_modules().iter() {
+            let resolutions = self.resolutions(module).borrow();
+            for (_, resolution) in resolutions.iter() {
+                let resolution = resolution.borrow();
+                if let Some(binding) = resolution.binding
+                && let Some(glob_binding) = resolution.shadowed_glob
+                && let NameBinding { kind: NameBindingKind::Import { import, .. }, .. } = glob_binding
+                && let Some(import_id) = import.id()
+                && let import_local_def_id = self.local_def_id(import_id)
+                && self.effective_visibilities.is_exported(import_local_def_id)
+                && glob_binding.vis.is_public()
+                && !binding.vis.is_public()
+                && let Some(def_id) = binding.res().opt_def_id()
```

Comment:
This condition is incorrect, you can shadow a glob import with something without a def id, e.g. with built-in type `use u8;`.